packages/doenetml-worker-rust/lib-doenetml-core/src/dast/ref_resolve/build_resolver.rs

```diff
@@ -333,32 +335,42 @@ impl Resolver {
 ///
 /// For each such attribute, the value of the name is determined from the attribute's children.
 /// The attribute must have exactly one text child to be considered valid.
-fn get_element_names_with_source(element: &FlatElement) -> Vec<NameWithSource> {
-    let name_re = Regex::new(r"^(source-(?<source_doc>\d+):)?name$").unwrap();
-    let names_with_source = element
-        .attributes
-        .iter()
-        .filter_map(|attr| match name_re.captures(&attr.name) {
-            None => None,
-            Some(caps) => {
-                let source_doc: SourceDoc = match caps.name("source_doc") {
-                    // if the attribute is `name`, then get `source_doc` from the element itself
-                    None => element.source_doc.into(),
-                    // if the attribute is `source-n:name`, where `n` is number, then `n` is the `source_doc`
-                    Some(source_match) => source_match.as_str().try_into().unwrap(),
-                };
-                match (attr.children.len(), attr.children.first()) {
-                    // A name attribute should have exactly one text child. Otherwise it is considered invalid.
-                    (1, Some(UntaggedContent::Text(name))) => Some(NameWithSource {
-                        name: name.to_string(),
-                        source_doc,
-                    }),
-                    _ => None,
-                }
+fn get_element_names_with_source(
+    element: &FlatElement,
+) -> impl Iterator<Item = NameWithSource> + use<'_> {
+    element.attributes.iter().filter_map(|attr| {
+        // If the attribute name is `name` or of the form `source-n:name` where `n` is an integer,
+        // then determine both the name (which is the attribute value)
+        // and the source doc
+        let source_doc: SourceDoc = if attr.name == "name" {
+            // for the attribute 'name`, the source doc is the element's source doc`
+            element.source_doc.into()
+        } else if attr.name.starts_with("source-") && attr.name.ends_with(":name") {
```

Comment:
Add a comment with a date that says we previously tried a regex here but it produced a major slowdown on [specify test]